core-primitives/stf-state-handler/src/in_memory_state_file_io.rs

```diff
@@ -27,42 +27,54 @@ use crate::{
 	state_snapshot_primitives::StateId,
 };
 use codec::Encode;
-use itp_types::ShardIdentifier;
-use std::{collections::HashMap, hash::Hasher as HasherTrait, vec::Vec};
+use itp_types::{ShardIdentifier, H256};
+use sp_core::blake2_256;
+use std::{boxed::Box, collections::HashMap, vec::Vec};
 
-type StateHash = u64;
+type StateHash = H256;
 type ShardDirectory<State> = HashMap<StateId, (StateHash, State)>;
 type ShardsRootDirectory<State> = HashMap<ShardIdentifier, ShardDirectory<State>>;
+type InnerStateSelector<State, ExternalState> =
+	Box<dyn Fn(ExternalState) -> State + Send + Sync + 'static>;
+type ExternalStateGenerator<State, ExternalState> =
```

Comment:
That's interesting. Didn't know that's possible

Reply:
Which part? Defining a `Box<dyn Fn(..)>`? This allows me to define a type alias for a function object.

Reply:
Yes, exactly :)